platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/eventhandling/EventTransactionPool.java

```diff
@@ -63,30 +68,34 @@ public class EventTransactionPool implements TransactionPool, TransactionSupplie
 
     protected final SettingsProvider settings;
 
-    // Used for creating spy objects for unit tests
-    public EventTransactionPool() {
-        settings = null;
-        inFreeze = null;
-    }
-
     /**
      * Creates a new transaction pool for transactions waiting to be put in an event.
      *
-     * @param settings
-     * 		settings to use
-     * @param inFreeze
-     * 		Indicates if the system is currently in a freeze
+     * @param metrics  the metrics engine
+     * @param settings settings to use
+     * @param inFreeze Indicates if the system is currently in a freeze
      */
-    public EventTransactionPool(final SettingsProvider settings, final BooleanSupplier inFreeze) {
+    public EventTransactionPool(
+            @NonNull final Metrics metrics,
+            @Nullable SettingsProvider settings,
+            @Nullable final BooleanSupplier inFreeze) {
         this.settings = settings;
         this.inFreeze = inFreeze;
+
+        metrics.getOrCreate(
```

Comment:
Null check

Reply:
This is dereferenced in the constructor and will throw an exception promptly if null.